client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/util/FileUtils.java

```diff
@@ -152,4 +162,47 @@ public static void replaceDir(Consumer<Path> dirContentUpdater, Path target, Pat
     FileUtils.mkdirs(target.getParent());
     FileUtils.moveDir(work, target);
   }
+  
+  /**
+   * On Windows, retries the provided IO operation a number of times, in an effort to make the operation succeed.
+   * 
+   * Operations that might fail on Windows are file & directory move, as well as file deletion. These failures 
+   * are typically caused by the virus scanner and/or the Windows Indexing Service. These services tend to open a file handle 
+   * on newly created files in an effort to scan their content.
+   * 
+   * @param runnable the runnable whose execution should be retried
+   */
+  private static void retry(IORunnable runnable) throws IOException {
+    for (int retries = isWindows()? 30: 1; retries > 0; retries--) {
+      if (retries > 1) {    	  
+	    try {
+	      runnable.run();
+	      break;
+	    } catch(AccessDeniedException e) {
+	      // Sleep a bit to give a chance to the virus scanner / Windows Indexing Service to release the opened file handle
+	      try {
+	        Thread.sleep(100);
+	      } catch (InterruptedException ie) {
+	    	// Nothing meaningful to be done here
+	      }
+	    }
+      } else {
+    	// Give it a one last chance, and this time do not swallow the exception
+        runnable.run();
+      }
+    }
+  }
+  
+  /**
+   * A simple check whether the underlying operating system is Windows. 
+   * @return true if the OS is Windows
+   */
+  private static boolean isWindows() {
+	  synchronized (FileUtils.class) {
```

Comment:
The value could be simply statically assigned to `windows`. There is no need to calculate it lazily since we know it will always be needed and it avoids to do the synchronization every time.

Reply:
Obvious in retrospective. Will fix it accordingly, thanks!